tests/pipeline_operations_test.py

```diff
@@ -1,14 +1,76 @@
 import unittest
 import pyspark
 
+import apache_beam as beam
+import apache_beam.testing.test_pipeline as test_pipeline
+from apache_beam.testing.util import assert_that
+from apache_beam.testing.util import equal_to
+
+from pipeline_dp import DataExtractors
 from pipeline_dp.pipeline_operations import SparkRDDOperations
 from pipeline_dp.pipeline_operations import LocalPipelineOperations
+from pipeline_dp.pipeline_operations import BeamOperations
 
 
 class PipelineOperationsTest(unittest.TestCase):
     pass
 
 
+class BeamOperationsTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.ops = BeamOperations()
+        cls.data_extractors = DataExtractors(
+            partition_extractor=lambda x: x[1],
+            privacy_id_extractor=lambda x: x[0],
+            value_extractor=lambda x: x[2])
+
+    def test_filter_by_key_must_not_be_none(self):
```

Comment:
tests in-memory vs join-based are almost duplicated. It's good to avoid duplication especially if it's not hard. A good thing that there is a good solution - test parametrization, namely each test might have 1 bool parameter, say in_memory, if true then to run in-memory and false join-based

We can use absl parametrization, there are a few exampels [here](https://github.com/abseil/abseil-py/blob/master/absl/testing/parameterized.py).

Could you please make them parametrizable with absl? LMK if you have any questions.

Reply:
Thanks for suggesting that. Done.